interfaces/builtin/dm_crypt.go

```diff
@@ -0,0 +1,101 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/snap"
+)
+
+const dmCryptSummary = `allows encryption of block storages`
+
+const dmCryptBaseDeclarationSlots = `
+  dm-crypt:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+const dmCryptBaseDeclarationPlugs = `
+  tee:
+    allow-installation: false
+    deny-auto-connection: true
+`
+
+// The type for this interface
+type dmCryptInterface struct{}
+
+const dmCryptConnectedPlugAppArmor = `
+# Allow mapper access
+/dev/mapper/control rw,
+/dev/dm-[0-9]* rw,
+# allow use of cryptsetup from core snap
+/{,usr/}sbin/cryptsetup ixr,
+# Mount points could be in /run/media/<user>/* or /media/<user>/*
+/run/systemd/seats/* r,
+/{,run/}media/{,**} rw,
+mount options=(ro,nosuid,nodev) /dev/dm-[0-9]* -> /{,run/}media/**,
+mount options=(rw,nosuid,nodev) /dev/dm-[0-9]* -> /{,run/}media/**,
+
+#  exec mount/umount to do the actual operations
+/{,usr/}bin/mount ixr,
+/{,usr/}bin/umount ixr,
+
+# mount/umount (via libmount) track some mount info in these files
+/run/mount/utab* wrlk,
+`
+
+const dmCryptConnectedPlugSecComp = `
+# Description: Allow kernel keyring manipulation
+add_key
+keyctl
+request_key
+`
+
+// dm-crypt Note also that
+// in many cases ppp_generic is statically linked into the kernel (CONFIG_DM_CRYPT=y)
```

Comment:
s/ppp_generic/dm_crypt/

Is this comment true for dm-crypt? I notice in hirsute at least we seem to have `CONFIG_DM_CRYPT=m`

Reply:
Apart of wrong module name yes, comment is true.
You are right, for generic ubuntu kernels it will be usually `CONFIG_DM_CRYPT=m`
But whenever enablement is done for given hw and we build custom kernel for the hw, then we usually set `CONFIG_DM_CRYPT=y` as disk encryption is must have feature anyway. And having it as loadable module would only add need to include the module in the initrd. For no clear gain.
Fixing description, to make it more clear.